diff -Nur zaptel-1.2.6/Makefile zaptel-1.2.6.as_needed-fix/Makefile
--- zaptel-1.2.6/Makefile	2006-08-02 04:55:50.000000000 +0200
+++ zaptel-1.2.6.as_needed-fix/Makefile	2006-08-02 04:55:51.000000000 +0200
@@ -231,7 +231,7 @@
 	ar rcs libtonezone.a $^
 
 $(LIBTONEZONE_SO): $(TZOBJS)
-	$(CC) $(LDFLAGS) -shared -Wl,-soname,$(LIBTONEZONE_SO).$(LIBTONEZONE_SO_MAJOR_VER).$(LIBTONEZONE_SO_MINOR_VER) -lm -o $@ $^
+	$(CC) $(LDFLAGS) -shared -Wl,-soname,$(LIBTONEZONE_SO).$(LIBTONEZONE_SO_MAJOR_VER).$(LIBTONEZONE_SO_MINOR_VER) -o $@ $^ -lm
 
 ztcfg.c: ztcfg.h
 
