diff -Nur zaptel-1.2.7.org/zaptel.c zaptel-1.2.7_/zaptel.c
--- zaptel-1.2.7.org/zaptel.c   2005-12-17 03:04:05.000000000 +0100
+++ zaptel-1.2.7_/zaptel.c      2006-08-07 15:36:18.487081500 +0200
 		} else {
 			if (copy_from_user(chan->writebuf[chan->inwritebuf], usrbuf, amnt))
 				return -EFAULT;
+                        if ((chan->flags & ZT_FLAG_HDLC) && chan->span->ioctl != NULL){
+                                if (chan->span->ioctl(chan, ZT_DCHAN_TX_V2, amnt)==0){
+                                        return amnt;
+                                }
+                        }
 			chan->writen[chan->inwritebuf] = amnt;
 		}
 		chan->writeidx[chan->inwritebuf] = 0;
diff -Nur zaptel-1.2.7.org/zaptel.h zaptel-1.2.7_/zaptel.h
--- zaptel-1.2.7.org/zaptel.h   2005-12-17 03:04:05.000000000 +0100
+++ zaptel-1.2.7_/zaptel.h      2006-08-07 15:36:18.479081000 +0200
@@ -632,6 +632,11 @@
  *  60-80 are reserved for private drivers
  *  80-85 are reserved for dynamic span stuff
  */
+#define ZT_DCHAN_TX    _IOR (ZT_CODE, 60, int)
+#define ZT_DCHAN_TX_V1  ZT_DCHAN_TX
+#define ZT_DCHAN_TX_V2  ZT_DCHAN_TX
+/*
+ */
 
 /*
  * Create a dynamic span
