diff -ur zaptel-orig/Makefile zaptel/Makefile
--- zaptel-orig/Makefile	2004-03-26 00:44:19.000000000 -0700
+++ zaptel/Makefile	2004-04-09 12:29:41.761752456 -0600
@@ -63,8 +63,8 @@
 
 linux26: 
 linux26: prereq $(BINS)
-	@if ! [ -d /usr/src/linux-2.6 ]; then echo "Link /usr/src/linux-2.6 to your kernel sources first!"; exit 1 ; fi
-	make -C /usr/src/linux-2.6 SUBDIRS=$(PWD) modules
+	@if ! [ -d /usr/src/linux-`uname -r` ]; then echo "Link /usr/src/linux-`uname -r` to your kernel sources first!"; exit 1 ; fi
+	make -C /usr/src/linux-`uname -r` SUBDIRS=$(PWD) modules
 
 obj-m := $(MODULESO)
 
@@ -263,35 +263,7 @@
 	install -m 644 torisa.h $(INSTALL_PREFIX)/usr/include/linux
 	install -m 644 tonezone.h $(INSTALL_PREFIX)/usr/include
 	( cd $(INSTALL_PREFIX)/usr/lib ; rm -f libtonezone.so ; ln -sf $(LIBTONEZONE) libtonezone.so )
-	/sbin/ldconfig
-	if [ -f $(MODCONF) ]; then mv -f $(MODCONF) $(MODCONF).bak ; fi
-	cat $(MODCONF).bak | grep -v "alias char-major-250" | \
-	grep -v "post-install torisa /sbin/ztcfg" | \
-	grep -v "post-install wcfxsusb /sbin/ztcfg" | \
-	grep -v "post-install wcfxs /sbin/ztcfg" > $(MODCONF) || true
-	if ! grep "options torisa" $(MODCONF); then \
-		echo "options torisa base=$(BASEADDR)" >> $(MODCONF); \
-	fi
-	if ! grep "alias char-major-196" $(MODCONF); then \
-		echo "alias char-major-196 $(PRIMARY)" >> $(MODCONF); \
-	fi
 		
-	for x in $(MODULES); do \
-		if ! grep "post-install $$x" $(MODCONF); then \
-			if ! grep "install $$x " $(MODCONF); then \
-				if [ "$$x" != "zaptel" ] ; then \
-					if [ -f zaptel.ko ]; then echo "install $$x /sbin/modprobe --ignore-install $$x && /sbin/ztcfg" >> $(MODCONF); \
-					else echo "post-install $$x /sbin/ztcfg" >> $(MODCONF); \
-					fi; \
-				fi; \
-			fi; \
-		fi; \
-	done
-
-	if [ -d /etc/modutils ]; then \
-		/sbin/update-modules ; \
-	fi
-	-/sbin/depmod -a
 	[ -f $(INSTALL_PREFIX)/etc/zaptel.conf ] || install -m 644 zaptel.conf.sample $(INSTALL_PREFIX)/etc/zaptel.conf
 
 config:
Only in zaptel: Makefile~
